src/BizHawk.Client.Common/ArgParser.cs

```diff
@@ -2,221 +2,227 @@
-				{
-					luaConsole = true;
-				}
-				else if (argDowncased.StartsWithOrdinal("--socket_port="))
-				{
-					var port = ushort.TryParse(arg.Substring(14), out var i1) ? i1 : (ushort) 0;
-					if (port > 0) socketPort = port;
-				}
-				else if (argDowncased.StartsWithOrdinal("--socket_ip="))
-				{
-					socketIP = argDowncased.Substring(argDowncased.IndexOf('=') + 1);
-				}
-				else if (argDowncased.StartsWithOrdinal("--socket_udp"))
-				{
-					socketProtocol = ProtocolType.Udp;
-				}
-				else if (argDowncased.StartsWithOrdinal("--mmf="))
-				{
-					mmfFilename = arg.Substring(arg.IndexOf('=') + 1);
-				}
-				else if (argDowncased.StartsWithOrdinal("--url_get="))
-				{
-					urlGet = arg.Substring(arg.IndexOf('=') + 1);
-				}
-				else if (argDowncased.StartsWithOrdinal("--url_post="))
-				{
-					urlPost = arg.Substring(arg.IndexOf('=') + 1);
-				}
-				else if (argDowncased.StartsWithOrdinal("--audiosync="))
-				{
-					audiosync = argDowncased.Substring(argDowncased.IndexOf('=') + 1) == "true";
-				}
-				else if (argDowncased.StartsWithOrdinal("--open-ext-tool-dll="))
-				{
-					// the first ext. tool from ExternalToolManager.ToolStripMenu which satisfies both of these will be opened:
-					// - available (no load errors, correct system/rom, etc.)
-					// - dll path matches given string; or dll filename matches given string with or without `.dll`
-					openExtToolDll = arg.Substring(20);
-				}
-				else if (argDowncased.StartsWithOrdinal("--userdata="))
-				{
-					userdataUnparsedPairs = new();
-					foreach (var s in arg.Substring(11).Split(';'))
-					{
-						var iColon = s.IndexOf(':');
-						if (iColon is -1) throw new ArgParserException("malformed userdata (';' without ':')");
-						userdataUnparsedPairs.Add((s.Substring(startIndex: 0, length: iColon), s.Substring(iColon + 1)));
-					}
-				}
-				else
+		private static readonly Argument<string?> ArgumentRomFilePath = new(name: "rom", () => null, description: "path; if specified, the file will be loaded the same way as it would be from `File` > `Open...`");
+
+		private static readonly IReadOnlyList<Option> GeneratedOptions;
+
+		private static readonly BespokeOption<string?> OptionAVDumpAudioSync = new(name: "--audiosync", description: "bool; `true` is the only truthy value, all else falsey; if not set, uses remembered state from config");
+
+		private static readonly BespokeOption<int?> OptionAVDumpEndAtFrame = new("--dump-length");
+
+		private static readonly BespokeOption<string?> OptionAVDumpFrameList = new(name: "--dump-frames", description: $"implies `--{OptionAVDumpEndAtFrame.Name}=<end>` where `<end>` is the highest frame listed");
+
+		private static readonly BespokeOption<string?> OptionAVDumpName = new("--dump-name");
+
+		private static readonly BespokeOption<bool> OptionAVDumpQuitWhenDone = new("--dump-close");
+
+		private static readonly BespokeOption<string?> OptionAVDumpType = new("--dump-type");
```

Comment:
This actually requires `--dump-name` to be specified as well, which should probably be documented. Otherwise this option is silently not applied.